bigframes/core/nodes.py

```diff
@@ -155,6 +155,7 @@ def __hash__(self):
 @dataclass(frozen=True)
 class ReadLocalNode(BigFrameNode):
     feather_bytes: bytes
+    session: bigframes.session.Session
 
```

Comment:
Local data is session-independent, we don't want to add a session constrain to the node. Don't worry about a dataframe/block not having session, that just means you can execute it anywhere, as all the data sources are local.

Reply:
The data itself is independent of session, yes. But when reading a local data, a specific session will be used. And when executing the query, we'd call that particular session. Do we have other options than keep a record here?

Reply:
A session will be used to execute the tree, yes, but the choice of session need not be constrained by the tree itself. You can check a tree to see if it has a required session, and otherwise, just use the default session to execute trees that don't depend on a specific session.

Reply:
Discussed offline, lets just make the session an optional field that we set when users have a specific session they used to read the local data with `session.read_gbq`.

Reply:
Updated.